lib/mpi/responses/add_parser.rb

```diff
@@ -44,6 +46,21 @@ def parse
         parse_ids(attributes)
       end
 
+      def error_details(mpi_codes)
+        error_details = {
+          ack_detail_code: @code,
+          id_extension: locate_element(@original_body, ACKNOWLEDGEMENT_TARGET_MESSAGE_ID_EXTENSION_XPATH),
+          error_texts: []
+        }
+        locate_elements(@original_body, ACKNOWLEDGEMENT_DETAIL_TEXT_XPATH).each do |node|
+          error_details[:error_texts].append(node.text) unless error_details[:error_texts].include?(node.text)
+        end
+        mpi_codes[:other] = [mpi_codes[:other].first.merge(error_details)]
```

Comment:
What if we put this in `mpi_codes[:error_details]`. Seems like it'd simplify this code and the mpi service part